samcli/lib/build/build_strategy.py

```diff
@@ -118,6 +118,7 @@ def build_single_function_definition(self, build_definition: FunctionBuildDefini
         # we should create a copy and pass it down, otherwise additional env vars like LAMBDA_BUILDERS_LOG_LEVEL
         # will make cache invalid all the time
         container_env_vars = deepcopy(build_definition.env_vars)
+        container_dir_mounts = deepcopy(build_definition.dir_mounts)
```

Comment:
Here in `build_single_function_definition` a copy of `build_definition.env_vars` is made.
Below however, in `build_single_layer_definition` there is no copy. Is this intended, or should it be fixed?